core-tests/src/test/java/org/mule/runtime/core/policy/AbstractPolicyProcessorTestCase.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.policy;
+
+import static java.util.UUID.randomUUID;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.api.message.Message;
+import org.mule.runtime.core.DefaultEventContext;
+import org.mule.runtime.core.api.Event;
+import org.mule.runtime.core.api.construct.FlowConstruct;
+import org.mule.runtime.core.api.processor.Processor;
+import org.mule.runtime.core.message.DefaultMessageBuilder;
+import org.mule.runtime.core.session.DefaultMuleSession;
+import org.mule.tck.message.StringAttributes;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+public abstract class AbstractPolicyProcessorTestCase {
+
+  private static final String INIT_VAR_NAME = "initVarName";
+  private static final String INIT_VAR_VALUE = "initVarValue";
+  private static final String ADDED_VAR_NAME = "addedVarName";
+  private static final String ADDED_VAR_VALUE = "addedVarValue";
+  private static final String PAYLOAD = "payload";
+
+  private static final Message MESSAGE = new DefaultMessageBuilder().attributes(new StringAttributes()).payload(PAYLOAD).build();
```

Comment:
Message.builder()  is returning Message.PayloadBuilder and does not let me set the attributes